### PR TITLE
chore: Update calico manifests to v3.4.0

### PR DESCRIPTION
**Pull request by song-jiang**
**Reason for Change**:

This PR updates calico manifests to v3.4.0

**Issue Fixed**:

None

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
None.

### CODE REVIEW

File: parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml [hunk 1]
```diff
@@ -127,7 +31,7 @@ data:
           "datastore_type": "kubernetes",
           "nodename": "__KUBERNETES_NODE_NAME__",
           "mtu": 1500,
-          "ipam": <calicoIPAMConfig>,
+          "ipam" : <calicoIPAMConfig>,
```
Comment (sylr) [inline parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: Unneeded whitespace.

**Comment by sylr** [comment]
Hi @song-jiang I found a lot of uneeded indenting in the yaml you updates, .i.e:

```
array:
- item
- item2
```

```
array:
  - item
  - item2
```

The dash `-` of the array items should stay at aligned with the name of the array.

File: parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml [hunk 2]
```diff
@@ -190,323 +93,212 @@ spec:
     metadata:
       labels:
         k8s-app: calico-typha
-        addonmanager.kubernetes.io/mode: "EnsureExists"
```
Comment (sylr) [inline parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: If you remove this the addon-manager won't create the object.

File: parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml (hunk 1 above)
Comment (song-jiang) [reply parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: Thanks @sylr for the review. The manifests has been auto generated for aks-engine from a calico manifest. I will fix the script to output a better format.

File: parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml (hunk 2 above)
Comment (song-jiang) [reply parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: From my test, I can see `calico-typha` pod running by just having label `addonmanager.kubernetes.io/mode: "EnsureExists"` set at `Deployment` metadata layer. This is similar to Deployment `calico-typha-horizontal-autoscaler` below.

So I am thinking omitting this label for `template.metadata` should be fine. WDYT?

**Comment by song-jiang** [comment]
@sylr Could you review it again? I have address all your comments.

File: parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml (hunk 2 above)
Comment (sylr) [reply parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: You're right 👍  I missed that it was not in the good section

File: parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml [hunk 3]
```diff
@@ -190,222 +93,86 @@ spec:
...
-    }
-
----
+# This manifest creates a Pod Disruption Budget for Typha to allow K8s Cluster Autoscaler to evict
 
-# Typha Horizontal Autoscaler Deployment
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
```
Comment (sylr) [inline parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: You should not include the PodDisruptionBudget object. It was added at a time but was removed cause it caused problem with the upgrade problem (see https://github.com/Azure/aks-engine/pull/300).

Also a PodDisruptionBudget can not be updated at the moment. It needs to be removed and recreated if you want to change the spec so it's better to not include it.

File: parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml [hunk 4]
```diff
@@ -435,155 +201,171 @@ spec:
...
     spec:
+      nodeSelector:
+        beta.kubernetes.io/os: linux
       hostNetwork: true
       tolerations:
-        # Allow the pod to run on the master.  This is required for
-        # the master to communicate with pods.
-        - effect: NoSchedule
-          operator: Exists
+        # Make sure calico-node gets scheduled on all nodes.
```
Comment (sylr) [inline parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: Please align all array items comments with the `-`

Comment (song-jiang) [reply parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: Fixing comment indentations by a script seems to be much harder. Let me try and come back to you later.

Comment (song-jiang) [reply parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml]: I fixed indentation manually. Will update script later. @sylr Please have another look. Thanks.

**Comment by song-jiang** [comment]
@sylr Could you help with another look?

**Comment by song-jiang** [comment]
@sylr Have you got chance to review my latest change?

**Comment by sylr** [comment]
@song-jiang sorry but I've been really overwhelmed at work recently. I don't think I'll have the time to review it more in the following days.

**Pull request closed without merging**